src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityQuantumStorage.java

```diff
@@ -126,13 +166,217 @@ public void onPlacement(@Nullable EntityLivingBase placer) {
+                                            BoolValue.Dynamic isActive,
+                                            IntSyncValue amountPerCycle,
+                                            IntSyncValue ticksPerCycle) {
+        return panel.height(209)
+                .bindPlayerInventory()
+                .child(Flow.column()
+                        .pos(7, 28)
+                        .crossAxisAlignment(Alignment.CrossAxis.START)
+                        .coverChildren()
+                        .child(IKey.lang("gregtech.creative." +
+                                (isTank ? "tank.mbpc" : "chest.ipc"))
+                                .asWidget()
+                                .marginBottom(2))
+                        .child(new TextFieldWidget()
+                                .left(2)
+                                .marginBottom(15)
+                                .size(154, 14)
+                                .keepScrollBarInArea(true)
+                                .setNumbers(1, Integer.MAX_VALUE)
+                                .setMaxLength(11)
+                                .value(amountPerCycle))
+                        .child(IKey.lang("gregtech.creative.tank.tpc").asWidget()
+                                .marginBottom(2))
+                        .child(new TextFieldWidget()
+                                .left(2)
+                                .size(154, 14)
+                                .keepScrollBarInArea(true)
+                                .setNumbers(1, Integer.MAX_VALUE)
+                                .setMaxLength(11)
+                                .value(ticksPerCycle)))
+                .child(new ToggleButton()
+                        .disableHoverBackground()
+                        .pos(7, 101)
+                        .size(162, 20)
+                        .overlay(IKey.lang(() -> isActive.getBoolValue() ?
+                                "gregtech.creative.activity.on" :
+                                "gregtech.creative.activity.off"))
+                        .value(new BooleanSyncValue(isActive::getBoolValue, value -> {
+                            isActive.setBoolValue(value);
+                            scheduleRenderUpdate();
+                            var c = getQuantumController();
+                            if (c != null) c.onHandlerUpdate();
+                        })));
+    }
+
+    protected boolean isVoiding() {
+        return this.voiding;
+    }
+
+    protected void setVoiding(boolean isVoiding) {
+        this.voiding = isVoiding;
+        if (!getWorld().isRemote) {
+            writeCustomData(UPDATE_IS_VOIDING, buf -> buf.writeBoolean(this.voiding));
+            markDirty();
+        }
+    }
+
+    protected boolean isLocked() {
+        return this.locked;
+    }
+
+    protected void setLocked(boolean locked) {
+        if (this.locked == locked) return;
+        this.locked = locked;
+        if (!getWorld().isRemote) {
```

Comment:
Do we need a write custom data here?

Reply:
All the other `setXXX` methods are using a custom data call, so maybe we do need it here.